common4j/src/main/com/microsoft/identity/common/java/util/ported/Pair.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.identity.common.java.util.ported;
```

Comment:
ported as in ported from Android to java? Do we already have something similar in JAVA that we can use?

Reply:
This requires including javafx and Java 9
https://docs.oracle.com/javase/9/docs/api/javafx/util/Pair.html

Reply:
ok, we certainly don't want that 😁

Reply:
We could substitute AbstractMap.SimpleEntry<U, V> (which implements Map.Entry) for this, rather than writing our own or needing to so this port.  A thought.

Reply:
@AdamBJohnsonx AbstractMap.SimpleEntry seems to be used only in SharedPrefFileManager, so that will never be ported to java-common.

Reply:
ah... it's the other way around.

Reply:
Looking into the code. I think I'll keep pair. AbstractMap.SimpleEntry would also work, but its naming (key, value) might confuse future readers.

Reply:
Don't really care, but it's another class.  Any pure-java developer is likely familiar with map-entry syntax, FWIW.